Mathlib/GroupTheory/Coset.lean

```diff
@@ -0,0 +1,899 @@
+
+#align subgroup.card_dvd_of_injective Subgroup.card_dvd_of_injective
+#align add_subgroup.card_dvd_of_injective AddSubgroup.card_dvd_of_injective
+
+@[to_additive]
+theorem card_dvd_of_le {H K : Subgroup α} [Fintype H] [Fintype K] (hHK : H ≤ K) : card H ∣ card K :=
+  card_dvd_of_injective (inclusion hHK) (inclusion_injective hHK)
+#align subgroup.card_dvd_of_le Subgroup.card_dvd_of_le
+#align add_subgroup.card_dvd_of_le AddSubgroup.card_dvd_of_le
+
+@[to_additive]
+theorem card_comap_dvd_of_injective (K : Subgroup H) [Fintype K] (f : α →* H) [Fintype (K.comap f)]
+    (hf : Function.Injective f) : Fintype.card (K.comap f) ∣ Fintype.card K := by
+  haveI : Fintype ((K.comap f).map f) :=
+      Fintype.ofEquiv _ (equivMapOfInjective _ _ hf).toEquiv;
+    calc
+      Fintype.card (K.comap f) = Fintype.card ((K.comap f).map f) :=
+        Fintype.card_congr (equivMapOfInjective _ _ hf).toEquiv
+      _ ∣ Fintype.card K := card_dvd_of_le (map_comap_le _ _)
+
+#align subgroup.card_comap_dvd_of_injective Subgroup.card_comap_dvd_of_injective
+#align add_subgroup.card_comap_dvd_of_injective AddSubgroup.card_comap_dvd_of_injective
+
+end Subgroup
+
+namespace QuotientGroup
+
+variable [Group α]
+
+/-- If `s` is a subgroup of the group `α`, and `t` is a subset of `α ⧸ s`, then there is a
+(typically non-canonical) bijection between the preimage of `t` in `α` and the product `s × t`. -/
+@[to_additive "If `s` is a subgroup of the additive group `α`, and `t` is a subset of `α ⧸ s`, then
+ there is a (typically non-canonical) bijection between the preimage of `t` in `α` and the product
+ `s × t`."]
+noncomputable def preimageMkEquivSubgroupTimesSet (s : Subgroup α) (t : Set (α ⧸ s)) :
+    QuotientGroup.mk ⁻¹' t ≃ s × t
+    where
+  toFun a :=
+    ⟨⟨((Quotient.out' (QuotientGroup.mk a)) : α)⁻¹ * a,
+        leftRel_apply.mp (@Quotient.exact' _ (leftRel s) _ _ <| Quotient.out_eq' _)⟩,
+      ⟨QuotientGroup.mk a, a.2⟩⟩
+  invFun a :=
+    ⟨Quotient.out' a.2.1 * a.1.1,
+      show QuotientGroup.mk _ ∈ t by
+        rw [mk_mul_of_mem _ a.1.2, out_eq']
+        exact a.2.2⟩
+  left_inv := fun ⟨a, ha⟩ => Subtype.eq <| show _ * _ = a by simp
+  right_inv := fun ⟨⟨a, ha⟩, ⟨x, hx⟩⟩ => by ext <;> simp [ha]
+#align quotient_group.preimage_mk_equiv_subgroup_times_set
+  QuotientGroup.preimageMkEquivSubgroupTimesSet
+#align quotient_add_group.preimage_mk_equiv_add_subgroup_times_set
+  QuotientAddGroup.preimageMkEquivAddSubgroupTimesSet
+
+end QuotientGroup
+
+-- porting note: this library note may need to be updated, but I'm not sure how it should be
+-- rewritten exactly.
+library_note "use CoeTCₓ"/--
+We use the class `CoeTCₓ` instead of `Coe` if the first argument is a variable,
+or if the second argument is a variable not occurring in the first.
+Using `Coe` would cause looping of type-class inference. See
+<https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/remove.20all.20instances.20with.20variable.20domain>
+-/
```

Comment:
```suggestion
```
I looked at all the reference to this library note in mathlib3 and I believe none of them apply in Lean 4, except for the `Sign` coercion.  For that particular coercion please look at the `coercion into rings` library note.

As a general answer to "what coercion class should I use", please refer to this guide:
https://github.com/leanprover/lean4/blob/38a0d1e3733e2ba65ec2d980f7567c568adf2145/src/Init/Coe.lean#L75-L88